streams/src/main/java/org/apache/kafka/streams/processor/assignment/TaskAssignmentException.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor.assignment;
+
+/**
+ * The TaskAssignmentException should be thrown when an error occurs during the assignment process
+ * that cannot be resolved automatically by the assignor alone.
+ */
+public class TaskAssignmentException extends RuntimeException {
```

Comment:
I think TaskAssignmentException already exists and is part of the public API, so we don't need to add it. All the exceptions for Streams are under `org.apache.kafka.streams.errors`

(btw: if you ever actually do need to introduce a new exception type, it should probably extend StreamsException. Kafka Streams exceptions all extend StreamsException, things in core or the clients will extend KafkaException or one of its subtypes, it's a whole thing)